docs/spec_draft.md

```diff
@@ -838,6 +839,74 @@ operation.
 
 [Back to Ops](#index-of-ops)
 
+## stablehlo.pad
+
+### Semantics
+
+Expands `operand` by padding around the array as well as between the elements of
+the array with the given `padding_value`.
+
+`edge_padding_low` and `edge_padding_high` specifies the amount of padding added
+at the low-end (next to index 0) and the high-end (next to the highest index) of
+each dimension respectively. The amount of padding can be negative, where the
+absolute value of negative padding indicates the number of elements to remove
+from the specified dimension.
+
+`interior_padding` specifies the amount of padding added between any two
+elements in each dimension which may not be negative. Interior padding occurs
+before edge padding such that negative edge padding will remove elements from
+the interior-padded operand.
+
+### Operands
+
+| Name                | Type                               |
+|---------------------|------------------------------------|
+| `operand`           | tensor of any supported types      |
+| `padding_value`     | tensor of any supported types      |
+| `edge_padding_low`  | 1-dimensional array of type `si64` |
+| `edge_padding_high` | 1-dimensional array of type `si64` |
+| `interior_padding`  | 1-dimensional array of type `si64` |
+
+### Results
+
+| Name     | Type                          |
+|----------|-------------------------------|
+| `result` | tensor of any supported types |
+
+### Constraints
+
+  * (C1) `operand` and `result` have the same element type.
+  * (C2) `operand` and `padding_value` have the same element type.
+  * (C3) `padding_value` should be rank 0.
+  * (C4) `edge_padding_low`, `edge_padding_high`, `interior_padding` have the
+  same rank as `operand`.
+  * (C5) `interior_padding` >= 0.
+  * (C6) Padding should not result in negative size for `result` dimensions.
```

Comment:
Can we infer the shape of the result as well?

Reply:
Can you clarify? The shape of the result is inferred in https://github.com/openxla/stablehlo/blob/main/stablehlo/dialect/StablehloOps.cpp#L4518.

Reply:
Ah sorry that I was not clear:
For ops like add, we are putting constraint on the shape of its result like "same shape as the operand". Can we add a constraint on the result of PadOp based on the edge and and inferior padding values for each dimension.

>>The shape of the result is inferred in

Yes that is right that we are already inferring that but IMO spec is SoT from where all those other artifacts (verifier/shape inference) should be inferred. 

For example, the dimension size  of the ith dimension of the `result` is given by 

`di + max(di - 1, 0 ) *  interior_padding[i] + edge_padding_low[i] + edge_padding_high[i]`

wdyt?

Reply:
Ah thanks for clarifying.

>> For ops like add, we are putting constraint on the shape of its result like "same shape as the operand". Can we add a constraint on the result of PadOp based on the edge and and inferior padding values for each dimension.

Since pad will fundamentally change the size of each dimension, shape can be different. However, you did bring up a good point since the ranks should remain unchanged for `result`. I wonder whether this is worth stating in (C4) since this information can be somewhat inferred and inherent from the op itself since it is "padding" dimensions not "adding" dimensions.

>>For example, the dimension size of the ith dimension of the result is given by
>>`di + max(di - 1, 0 ) * interior_padding[i] + edge_padding_low[i] + edge_padding_high[i]`, where di is the dimension size of ith dimension of the operand`

Makes sense!

Reply:
I like your last suggestion. However, we will still need to keep the constraint that `dim(result, i)` cannot be negative since the formula will still allow negative dimensions. 

How about:
0 $\le$ `dim(result, i)` for all `i`th dimension of `operand`, where `dim(result, i) = di + max(di - 1, 0) * interior_padding[i] + edge_padding_low[i] + edge_padding_high[i]` and `di = dim(operand, i)`.

Reply:
+1